db/model/roomType.js

```diff
@@ -69,13 +74,170 @@ module.exports = function roomType(seq, dataTypes) {
         assoc.type = RoomType.hasMany(models.Room, {
           foreignKey: 'type',
         });
+        assoc.bots = RoomType.belongsToMany(models.Bot, {
+          foreignKey: 'roomTypeId',
+          through: 'RoomTypeBots',
+        });
         assoc.writers = RoomType.belongsToMany(models.User, {
           as: 'writers',
           through: 'RoomTypeWriters',
           foreignKey: 'roomTypeId',
         });
       },
     },
+    hooks: {
+
+      /**
+       * Ensures that all bots in request actually exist
+       *
+       * @param {Instance} inst - The instance being created
+       * @returns {Promise} which resolves to the instance, or rejects if
+       *  a bot is not found or duplicate bots are requested
+       */
+      beforeCreate(inst /* , opts */) {
+        const bots = inst.dataValues.bots;
+
+        return new seq.Promise((resolve, reject) => {
+          if (bots == null) {
+            resolve(inst);
+          }
+
+          if (bots.length > new Set(bots).size) {
+            reject(new Error(`Cannot have duplicate bots`));
+          }
+
+          bots.map((botName, index) => {
+            seq.models.Bot.findOne({ where: { name: botName } })
+            .then((o) => {
+              if (o === null) {
+                reject(new Error(`Bot ${botName} not found`));
+              }
+
+              if (index === bots.length - 1) {
+                resolve(inst);
+              }
+
+            });
+          });
+        });
+      }, // hooks.beforeCreate
+
+      /**
+       * Ensures that all bots in request actually exist
+       *
+       * @param {Instance} inst - The instance being created
+       * @returns {Promise} which resolves to the instance, or rejects if
+       *  a bot is not found or duplicate bots are requested
+       */
+      beforeUpdate(inst /* , opts */) {
+        const bots = inst.dataValues.bots;
+
+        return new seq.Promise((resolve, reject) => {
```

Comment:
you're repeating a lot of the same stuff from your beforeCreate hook -- please pull out to a function which can be called from either hook